jsaddle/src/Language/Javascript/JSaddle/Types.hs

```diff
@@ -758,18 +758,29 @@ instance MonadError JSVal JSM where
     tries <- JSM $ asks _jsContextRef_tries
     sendReqsBatchVar <- JSM $ asks _jsContextRef_sendReqsBatchVar
     finishVar <- JSM $ liftIO newEmptyMVar
+    asyncExceptionVar <- JSM $ liftIO newEmptyMVar
     JSM $ liftIO $ atomically $ modifyTVar' tries $ M.insert tryId finishVar
     env <- JSM ask
     let end = sendReq Req_FinishTry
-    tryResult <- JSM $ liftIO $ withAsync (runReaderT (unJSM (a <* end)) $ env { _jsContextRef_myTryId = tryId }) $ \aa -> do
-      --IDEA: Continue speculatively executing forward
+        action = (runReaderT (unJSM (a <* end)) $ env { _jsContextRef_myTryId = tryId })
+          `catch` (\e@(SomeException _) -> do
+                      case fromException e of
+                        Just (_ :: AsyncCancelled) -> pure ()
```

Comment:
This took me a second to get, but I think I understand what's going on.

Reply:
This can introduce subtle bugs if the `AsyncCancelled` is captured here like this, and the user used it to cancel some thread. Instead I will add custom exception to this logic and rethrow the `AsyncCancelled` as it is.

Reply:
Got rid of this